### PR TITLE
Revive DefaultAssemblyProviderTests

### PR DESCRIPTION
**Pull request by pranavkm**
Partial fix for https://github.com/aspnet/Mvc/issues/4140

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc/project.json [hunk 1]
```diff
@@ -11,7 +11,8 @@
     "nowarn": [
       "CS1591"
     ],
-    "xmlDoc": true
+    "xmlDoc": true,
+    "preserveCompilationContext": true
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc/project.json]: cc @pakrym \ @davidfowl. Is this a terrible thing to do?

**Comment by pranavkm** [comment]
cc @NTaylorMullen

File: src/Microsoft.AspNetCore.Mvc/project.json (hunk 1 above)
Comment (pakrym) [inline src/Microsoft.AspNetCore.Mvc/project.json]: Yes, why are you doing it anyway?

Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc/project.json]: We have a test that verifies if a list of package names we have is near identical to this assembly's dependencies. In the past we used the `ILibraryManager` to discover these dependencies.

File: test/Microsoft.AspNetCore.Mvc.Core.Test/Infrastructure/DefaultAssemblyProviderTests.cs [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        }
+
+        private static RuntimeLibrary GetLibrary(string name, params string[] dependencyNames)
+        {
+            var dependencies = dependencyNames?.Select(d => new Dependency(d, "1.0.0")) ?? new Dependency[0];
+
+            return new RuntimeLibrary(
+                "package",
+                name,
+                "1.0.0",
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/Infrastructure/DefaultAssemblyProviderTests.cs]: Anyway we can grab the version in a way that wont require code updates every time we ship?

Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/Infrastructure/DefaultAssemblyProviderTests.cs]: We don't look at versions at all in our provider. I could change it to some random version if that makes it easier to grock.

File: src/Microsoft.AspNetCore.Mvc/project.json (hunk 1 above)
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc/project.json]: I guess I can add this flag to the test and exclude any test assembly.

File: test/Microsoft.AspNetCore.Mvc.Core.Test/Infrastructure/DefaultAssemblyProviderTests.cs (hunk 2 above)
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/Infrastructure/DefaultAssemblyProviderTests.cs]: Ya make it random so it doesn't raise questions

**Comment by NTaylorMullen** [comment]
:shipit:

**Pull request closed without merging**